src/app/interfaces/strongRoute.ts

```diff
@@ -192,18 +221,37 @@ export class StrongRoute {
    * Example output: "/home/house"
    */
   public toString(): string {
-    return "/" + (this.fullRoute ? this.fullRoute : "");
+    // Prevent double '/' at start of route string
+    const path = this.toRoute().join("/");
+    return path.length >= 2 ? path.substr(1) : path;
   }
 
   /**
-   * Router representation of the route
-   * ! This will use a path relative to the current page if directly
-   * inserted into the [routerLink] directive
+   * Router representation of the route inserted into the
+   * [routerLink] directive
    *
-   * Example output: ["home", "house"]
+   * Example output: ["/", "home", "house"]
```

Comment:
Add:

Outputs in the routerlink @Input format, i.e. an array of args passed to `Router#createUrlTree` via `routerLink`. See more https://angular.io/api/router/RouterLink

example output: `['/', 'projects', ':projectId', 'sites', ':siteid']`

to use queryPararms call the `queryParams` method

Reply:
Note there's nothing we can do here to validate if query params are being used or not.

When a user uses this in a routerLink, the routerLink args are passed to `@Input` `routerLinks` and the query params are passed separately to `queryParams`. These are two separate options and we can't control/validate that they both occur in the template

Reply:
What we could do is make our own `strongRouterLink` which ensures QSPs are provided or else fails.

We could also advise to always use `format` to ensure QSPs are formatted